homeassistant/components/template/weather.py

```diff
@@ -452,3 +468,248 @@ def _validate_forecast(
+            self._attr_supported_features |= WeatherEntityFeature.FORECAST_HOURLY
+        if config.get(CONF_FORECAST_TWICE_DAILY_TEMPLATE):
+            self._attr_supported_features |= WeatherEntityFeature.FORECAST_TWICE_DAILY
+
+        for key in (
+            CONF_APPARENT_TEMPERATURE_TEMPLATE,
+            CONF_CLOUD_COVERAGE_TEMPLATE,
+            CONF_DEW_POINT_TEMPLATE,
+            CONF_FORECAST_DAILY_TEMPLATE,
+            CONF_FORECAST_HOURLY_TEMPLATE,
+            CONF_FORECAST_TWICE_DAILY_TEMPLATE,
+            CONF_OZONE_TEMPLATE,
+            CONF_PRESSURE_TEMPLATE,
+            CONF_VISIBILITY_TEMPLATE,
+            CONF_WIND_BEARING_TEMPLATE,
+            CONF_WIND_GUST_SPEED_TEMPLATE,
+            CONF_WIND_SPEED_TEMPLATE,
+        ):
+            if isinstance(config.get(key), template.Template):
+                self._to_render_simple.append(key)
+                self._parse_result.add(key)
+
+    async def async_added_to_hass(self) -> None:
+        """Restore last state."""
+        await super().async_added_to_hass()
+        if (
+            (state := await self.async_get_last_state())
+            and state.state is not None
+            and state.state not in (STATE_UNKNOWN, STATE_UNAVAILABLE)
+            and (weather_data := await self.async_get_last_weather_data())
+        ):
+            self._rendered[
+                CONF_APPARENT_TEMPERATURE_TEMPLATE
+            ] = weather_data.last_apparent_temperature
+            self._rendered[
+                CONF_CLOUD_COVERAGE_TEMPLATE
+            ] = weather_data.last_cloud_coverage
+            self._rendered[CONF_CONDITION_TEMPLATE] = state.state
+            self._rendered[CONF_DEW_POINT_TEMPLATE] = weather_data.last_dew_point
+            self._rendered[CONF_HUMIDITY_TEMPLATE] = weather_data.last_humidity
+            self._rendered[CONF_OZONE_TEMPLATE] = weather_data.last_ozone
+            self._rendered[CONF_PRESSURE_TEMPLATE] = weather_data.last_pressure
+            self._rendered[CONF_TEMPERATURE_TEMPLATE] = weather_data.last_temperature
+            self._rendered[CONF_VISIBILITY_TEMPLATE] = weather_data.last_visibility
+            self._rendered[CONF_WIND_BEARING_TEMPLATE] = weather_data.last_wind_bearing
+            self._rendered[
+                CONF_WIND_GUST_SPEED_TEMPLATE
+            ] = weather_data.last_wind_gust_speed
+            self._rendered[CONF_WIND_SPEED_TEMPLATE] = weather_data.last_wind_speed
+
+    @property
+    def condition(self) -> str | None:
+        """Return the current condition."""
+        return self._rendered.get(CONF_CONDITION_TEMPLATE)
+
+    @property
+    def native_temperature(self) -> float | None:
+        """Return the temperature."""
+        return vol.Any(vol.Coerce(float), None)(
+            self._rendered.get(CONF_TEMPERATURE_TEMPLATE)
+        )
+
+    @property
+    def humidity(self) -> float | None:
+        """Return the humidity."""
+        return vol.Any(vol.Coerce(float), None)(
```

Comment:
Where is the voluptuous exception caught? We normally don't use voluptuous validators outside config validation or a context that handles those specific exceptions.